app/src/main/java/com/vishaltelangre/nerdcalci/core/MathEngine.kt

```
@@ -187,80 +93,105 @@ object MathEngine {
-            } catch (e: Exception) {
-                line.copy(result = "Err")
+    /** Format a numeric result for display. */
+    private fun formatResult(value: Double): String {
+        return if (value % 1.0 == 0.0) {
+            when {
+                value >= Int.MIN_VALUE && value <= Int.MAX_VALUE ->
+                    value.toInt().toString()
+                value >= Long.MIN_VALUE && value <= Long.MAX_VALUE ->
+                    value.toLong().toString()
+                else ->
+                    String.format("%.2e", value)
```

detekt (1.23.8)

[warning] String.format("%.2e", value) uses implicitly default locale for string formatting.

(detekt.potential-bugs.ImplicitDefaultLocale)